client/blocks/login/continue-as-user.jsx

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+import { parse as qsParse } from 'qs';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import { getCurrentUser } from 'state/current-user/selectors';
+import Gravatar from 'components/gravatar';
+
+/**
+ * Style dependencies
+ */
+import './continue-as-user.scss';
+
+function ContinueAsUser( { currentUser } ) {
+	const translate = useTranslate();
+	const [ validated, setValidated ] = useState( null );
+
+	useEffect( () => {
+		async function runEffect() {
+			if ( ! currentUser ) {
+				return null;
+			}
+
+			// TODO (sgomes): Replace with URLSearchParams when polyfilled (see #35408).
+			// const query = new URLSearchParams( window.location.search );
+			// const redirectUrl = query.get( 'redirect_to' );
+			const query = qsParse( window.location.search, { ignoreQueryPrefix: true } );
+			const redirectUrl = query.redirect_to;
```

Comment:
I'm not sure if my comment got lost somewhere, so I'll repeat it: the query parameters are in Redux: `state.ui.route.query.current` and can be retrieved with the `getCurrentQueryArguments` selector.

I'm not sure what the `ignoreQueryPrefix` arg does though.

Reply:
I don't know if I missed an earlier comment of yours (sorry if that's the case!), but this is the first time I'm reading this. That's very handy, thank you! Done, and I also simplified and clarified the effect logic a bit.

The `ignoreQueryPrefix` option is just a `qs` thing; all it does is ignore the leading `?` in the query string if it's there. Glad to not have to use it 🙂

Reply:
> I don't know if I missed an earlier comment of yours (sorry if that's the case!)

I couldn't find it either when searching for it later, so it probably ended up in an unsubmitted `textarea` in a closed tab 🙂